robotoff/prediction/ocr/packaging.py

```diff
@@ -0,0 +1,217 @@
+from typing import Optional, Union
+
+import cachetools
+from lark import Discard, Lark, Transformer
+
+from robotoff import settings
+from robotoff.prediction.ocr.dataclass import OCRResult, get_text
+from robotoff.prediction.ocr.grammar import (
+    generate_terminal_symbols_file,
+    normalize_string,
+)
+from robotoff.prediction.types import Prediction, PredictionType
+from robotoff.taxonomy import TaxonomyType
+from robotoff.types import PackagingElementProperty
+from robotoff.utils import load_json
+from robotoff.utils.text import strip_consecutive_spaces
+
+# Set of shapes to exclude when the material or recycling instructions were
+# not detected along the shape
+# Avoid many false positive
+SHAPE_ONLY_EXCLUDE_SET = {
+    "en:tray",  # coque
+    "en:packaging",  # emballage
+    "en:backing",  # plaque
+    "en:packet",  # pack
+    "en:string",  # ficelle
+    "en:bowl",  # bol
+    "en:sheet",  # feuilles
+    "en:net",  # filet
+    "en:pouch-flask",  # pouch, english word
+    "en:spoon",  # most of the spoon are false positive
+    "en:tablespoon",
+    "en:mold",  # moule
+    None,
+}
+
+
+def generate_packaging_lark_file(lang: str):
+    generate_terminal_symbols_file(
+        output_path=settings.GRAMMARS_DIR / f"terminal_packaging_shape_{lang}.lark",
+        lang=lang,
+        terminal_name="PACKAGING_SHAPES",
+        taxonomy_type=TaxonomyType.packaging_shape,
+        terminal_priority=1,
+        ignore_ids={"en:unknown"},
+    )
+    generate_terminal_symbols_file(
+        output_path=settings.GRAMMARS_DIR / f"terminal_packaging_material_{lang}.lark",
+        lang=lang,
+        terminal_name="PACKAGING_MATERIALS",
+        taxonomy_type=TaxonomyType.packaging_material,
+        terminal_priority=1,
+        ignore_ids={"en:unknown"},
+    )
+
+
+@cachetools.cached(cachetools.LRUCache(maxsize=5))
+def load_grammar(lang: str, start: str = "value", **kwargs) -> Lark:
+    return Lark.open(
+        str(settings.GRAMMARS_DIR / f"packaging_{lang}.lark"),
+        start=start,
+        **kwargs,
+    )
+
+
+class PackagingFRTransformer(Transformer):
+    def __init__(
+        self,
+        taxonomy_maps: dict[str, dict[str, list[str]]],
+        visit_tokens: bool = True,
+    ) -> None:
+        super().__init__(visit_tokens)
+        self.taxonomy_maps = taxonomy_maps
+
+    def value(self, items: list):
+        return items
+
+    def packaging(self, items: list):
+        data = {}
+        for item in items:
+            data.update(item)
+        return data
```

Comment:
why not: 
```suggestion
        return dict(items)
```

Reply:
It doesn't work, as each item needs to be a (k, v) tuple